protozoo/shared_config.py

```diff
@@ -0,0 +1,128 @@
+import copy
+import yaml
+
+from dataclasses import fields, InitVar, dataclass, field, replace, is_dataclass, _is_dataclass_instance
+from enum import Enum
+from pathlib import Path
+from typing import Optional, Callable, Any
+
+HIDE = "hide"
+
+
+class MapTo(Enum):
```

Comment:
What is the reason for having two modes? Can't we go from config to dict first, and then from dict to classes (and vice versa)?

Reply:
The idea to have both modes (now: Representations) was to include how to map from one to the other in the meta data of each field.
Instead of having two separate parsers that need to be kept in sync, each field defines it's conversions.